Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectGenerator/Scripts/MSBuildTools.cs

```diff
@@ -327,6 +337,28 @@ private static void RefreshGeneratedOutput(bool forceGenerateEverything, bool fo
             }
         }
 
+        private static Version GetCurrentTokenVersion()
+        {
+            string[] file = Directory.GetFiles(Path.Combine(Utilities.ProjectPath, "Temp"), "*_token.msb4u", SearchOption.TopDirectoryOnly);
```

Comment:
I don't think that Directory.GetFiles returns files in a deterministic sort order (at least, not according to: https://docs.microsoft.com/en-us/dotnet/api/system.io.directory.getfiles?view=netframework-4.8 see remarks)

I'm guessing most of the time we'll have a single token so it doesn't matter, but otherwise it will just return some arbitrary one.

Reply:
If we have many, we clearly are in a weird state, better to regen.